src/os/im/action/importactionmanager.js

```diff
@@ -351,24 +358,74 @@ os.im.action.ImportActionManager.prototype.getEntryItems = function(type) {
 
 /**
  * Executes enabled import action entries of a type against a set of items.
- *
  * @param {string} entryType The entry type.
- * @param {Array<T>=} opt_items The items to process.
+ * @param {Array<T>} items The items to process.
  * @param {boolean=} opt_unprocess Reset existing items
+ * @param {boolean=} opt_unprocessOnly Do not process enabled entries
+ * @private
  */
-os.im.action.ImportActionManager.prototype.processItems = function(entryType, opt_items, opt_unprocess) {
-  var items = opt_items || this.getEntryItems(entryType);
+os.im.action.ImportActionManager.prototype.processItemsInternal_ = function(
+    entryType,
+    items,
+    opt_unprocess,
+    opt_unprocessOnly) {
   if (items && items.length > 0) {
+    var promises = [];
     var entries = this.actionEntries[entryType];
     if (entries && entries.length > 0) {
       for (var i = 0; i < entries.length; i++) {
-        if (entries[i].isEnabled()) {
-          entries[i].processItems(items);
-        } else if (opt_unprocess) {
-          entries[i].unprocessItems(items);
+        var ps = null;
+        if (!opt_unprocessOnly && entries[i].isEnabled()) {
+          ps = entries[i].processItems(items);
+        } else if (opt_unprocess || opt_unprocessOnly) {
+          ps = entries[i].unprocessItems(items);
+        }
+        if (ps) {
+          ps.forEach((p) => {
+            promises.push(p);
+          });
         }
       }
     }
+
+    // once all processItems() are done, do a big notify of style and color changes
+    if (promises.length > 0) {
+      Promise.allSettled(promises).then((results) => {
```

Comment:
Use Promise.all() instead for old-browser compatibility; https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise